### PR TITLE
A few code quality improvements for DebugViewWriter

### PR DESCRIPTION
**Pull request by bartdesmet**
Addresses some plain silly inefficiencies in `DebugViewWriter` code. There's more, such as redundant use of `string.Format` using the current culture for simply string concatenations, but here I'm sticking with the trivial things. This is debug-time stuff anyway, only invoked through visualizers and IDE experiences. We can do more here later.

CC @VSadov

### CODE REVIEW

File: src/System.Linq.Expressions/src/System/Linq/Expressions/DebugViewWriter.cs [hunk 1]
```diff
@@ -303,7 +299,6 @@ private void VisitExpressions<T>(char open, char separator, IList<T> expressions
                 case '(': close = ')'; break;
                 case '{': close = '}'; break;
                 case '[': close = ']'; break;
-                case '<': close = '>'; break;
```
Comment (stephentoub) [inline src/System.Linq.Expressions/src/System/Linq/Expressions/DebugViewWriter.cs]: This was removed as dead code / an invalid open?

**Comment by stephentoub** [comment]
LGTM

File: src/System.Linq.Expressions/src/System/Linq/Expressions/DebugViewWriter.cs (hunk 1 above)
Comment (bartdesmet) [inline src/System.Linq.Expressions/src/System/Linq/Expressions/DebugViewWriter.cs]: Dead code indeed. You can check for `VisitExpressions` calls with '<'; none exist.

**Comment by bartdesmet** [comment]
Test Innerloop Ubuntu14.04 Release Build and Test please (build timeout @ http://dotnet-ci.cloudapp.net/job/dotnet_corefx/job/master/job/ubuntu14.04_release_prtest/27/)

**Pull request merged**